app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +136,183 @@ object DialogUtils {
             }
         }
     }
+
+    fun colorPickerBottomSheet(
+        context: Context,
+        previousColor: FloatArray,
+        /**
+         * Lambda that provides the string of the current color.
+         * Returns nothing. Used to save the hex value of the current color.
+         */
+        onSubmit: ((FloatArray?, String?) -> Unit)
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val view = LayoutInflater.from(context).inflate(R.layout.color_dialog, null)
+        val huePicker = view.findViewById<HueView>(R.id.huePicker)
+        val gradientPicker = view.findViewById<GradientView>(R.id.gradientPicker)
+        val hexText = view.findViewById<ForcableEditText>(R.id.hexText)
+        val hueText = view.findViewById<ForcableEditText>(R.id.hueText)
+        val satText = view.findViewById<ForcableEditText>(R.id.satText)
+        val valText = view.findViewById<ForcableEditText>(R.id.valText)
+        val redText = view.findViewById<ForcableEditText>(R.id.redText)
+        val greenText = view.findViewById<ForcableEditText>(R.id.greenText)
+        val blueText = view.findViewById<ForcableEditText>(R.id.blueText)
+        val topColorSquare = view.findViewById<View>(R.id.topColor)
+        val bottomColorSquare = view.findViewById<View>(R.id.bottomColor)
+        val confirmButton = view.findViewById<Button>(R.id.okButton)
+        confirmButton.setOnClickListener {
+            val submitArray = FloatArray(3)
+            submitArray[0] = (hueText.text.toString().toFloat())
+            submitArray[1] = (satText.text.toString().toFloat() / 100)
+            submitArray[2] = (valText.text.toString().toFloat() / 100)
+            onSubmit.invoke(submitArray, hexText.text.toString())
+            sheet.dismiss()
+        }
+        val cancelButton = view.findViewById<Button>(R.id.nopeButton)
+        cancelButton.setOnClickListener {
+            onSubmit.invoke(null, null)
+            sheet.dismiss()
+        }
+        sheet.setOnCancelListener {
+            onSubmit.invoke(null, null)
+            sheet.dismiss()
+        }
+        /**
+         * Function that updates all the views in the gradient.
+         * Takes in a color state data class to update views.
+         */
+        fun forceViewUpdates(colorState: ColorUtils.ColorState) {
+            val displaySat = (colorState.saturation * 100).toInt()
```

Comment:
shouldn't these be floats.

Reply:
They are floats, but when shown to the user in Photoshop they are shown as integer values between 0 and 100. I'll ask Nic if he wants us to display decimal values here or not, since that would be different from how photoshop handles this.

Reply:
Nic will give us direction on this later